python/lsst/ip/diffim/imageDecorrelation.py

```diff
@@ -164,74 +172,100 @@ def run(self, templateExposure, exposure, subtractedExposure, psfMatchingKernel)
         self.log.info("Convolving.")
-        correctedExposure, corrKern = DecorrelateALKernelTask._doConvolve(subtractedExposure, fcorrKernel)
+        correctedExposure, corrKern = DecorrelateALKernelTask._doConvolve(subtractedExposure, corrKernel)
         self.log.info("Updating correctedExposure and its PSF.")
 
         # Compute the subtracted exposure's updated psf
         psf = subtractedExposure.getPsf().computeImage().getArray()
-        psfc = DecorrelateALKernelTask.computeCorrectedDiffimPsf(fcorrKernel, psf, sig1=sig1, sig2=sig2)
+        psfc = DecorrelateALKernelTask.computeCorrectedDiffimPsf(corrKernel, psf, var1=var1, var2=var2)
         psfcI = afwImage.ImageD(psfc.shape[0], psfc.shape[1])
         psfcI.getArray()[:, :] = psfc
         psfcK = afwMath.FixedKernel(psfcI)
         psfNew = measAlg.KernelPsf(psfcK)
         correctedExposure.setPsf(psfNew)
         self.log.info("Complete.")
 
+        statObj = afwMath.makeStatistics(correctedExposure.getMaskedImage().getVariance(),
+                                         correctedExposure.getMaskedImage().getMask(),
+                                         afwMath.MEANCLIP, self.statsControl)
+        var = statObj.getValue(afwMath.MEANCLIP)
+        self.log.info("Variance (corrected diffim): %f" % var)
+
         return pipeBase.Struct(correctedExposure=correctedExposure, correctionKernel=corrKern)
 
     @staticmethod
-    def _computeDecorrelationKernel(kappa, sig1=0.2, sig2=0.2):
+    def _computeDecorrelationKernel(kappa, var1=0.04, var2=0.04):
         """! Compute the Lupton/ZOGY post-conv. kernel for decorrelating an
         image difference, based on the PSF-matching kernel.
         @param kappa  A matching kernel 2-d numpy.array derived from Alard & Lupton PSF matching
-        @param sig1   Average sqrt(variance) of template image used for PSF matching
-        @param sig2   Average sqrt(variance) of science image used for PSF matching
+        @param var1   Average variance of science image used for PSF matching
+        @param var2   Average variance of template image used for PSF matching
         @return a 2-d numpy.array containing the correction kernel
 
         @note As currently implemented, kappa is a static (single) kernel.
         """
+        kappa = DecorrelateALKernelTask._fixOddKernel(kappa)
         kft = fft2(kappa)
-        kft = np.sqrt((sig1**2 + sig2**2) / (sig1**2 + sig2**2 * kft**2))
+        kft = np.sqrt((var1 + var2) / (var1 + var2 * kft**2))
         pck = ifft2(kft)
         pck = ifftshift(pck.real)
+        fkernel = DecorrelateALKernelTask._fixEvenKernel(pck)
 
         # I think we may need to "reverse" the PSF, as in the ZOGY (and Kaiser) papers...
         # This is the same as taking the complex conjugate in Fourier space before FFT-ing back to real space.
         if False:  # TBD: figure this out. For now, we are turning it off.
-            pck = pck[::-1, :]
```

Comment:
I understand where it comes from in C, but I **very strongly** advocate that we stop using it in python, especially since `if False:` and `#if 0` actually have quite different behavior (interpreted vs. compiled language). See my community post above for details (and we can continue the conversation there, to not clutter David's PR).